app/controllers/actions/IdentifierAction.scala

```diff
@@ -39,19 +39,22 @@ class AuthenticatedIdentifierAction @Inject()(
                                              )
                                              (implicit val executionContext: ExecutionContext) extends IdentifierAction with AuthorisedFunctions {
 
-  private val enrolmentKey: String = "DAC6"
+  private val enrolmentKey: String = "HMRC-DAC6-ORG"
+  private val enrolmentIDKey: String = "DAC6ID"
 
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
-    authorised().retrieve(Retrievals.internalId and Retrievals.authorisedEnrolments) {
+    authorised(Enrolment(enrolmentKey)).retrieve(Retrievals.internalId and Retrievals.authorisedEnrolments) {
+
       case Some(internalID) ~ Enrolments(enrolments) =>
         val enrolmentID = {
           (for {
             enrolment <- enrolments.find(_.key.equals(enrolmentKey))
-            enrolmentIdentifier <- enrolment.getIdentifier(enrolmentKey)
-          } yield enrolmentIdentifier.value).getOrElse("EnrolmentID")
+            enrolmentIdentifier <- enrolment.getIdentifier(enrolmentIDKey)
+          } yield enrolmentIdentifier.value)
+            .getOrElse(throw new Exception("EnrolmentID Required for DAC6"))
```

Comment:
I don't think this is the right way. If the enrollment doesnt exist it'll be handed back an InsufficientEnrolments exception. How does this fail in testing? We should directing people to the registration service if they don't have an enrolment

Reply:
currently technical difficulties page, will talk with others to work out what is appropriate